balancer/frontend-v2
# Auto bump master when merging hotfixes. Push to deploy branch for deployment.

# Description

Original: #3354

New Changes: I've added another action to force update the `deploy` branch to the latest commit after bumping. Then we can make Cloudflare deploy this branch instead of master, so that there is only ever one deploy per merge. 

## Original Description

We have many hotfixes coming in each day to add new pools / gauges to the allowlist. Each of these currently requires a patch version bump, but this means as soon as one is merged all the remainder have git conflicts.

Instead of requiring the version bump in the PR, auto-bump the version on merge to master. This way we can easily merge all these hotfixes without constantly having to rebase and fix them.

This plugin will not bump weekly releases because it ignores bumping if any of the merged commits already include a version bump (it does a regex search for \d.\d.\d as the commit message in any of the commits)

The auto-tag has been left in to auto-tag weekly releases. 

I've tested this on a fork of this repo and you can see how it works doing a weekly release and a hotfix patch here: https://github.com/timjrobinson/frontend-v2/commits/master (Look at just the most recent 6 commits, the ones prior were testing)

The only downside to this is develop will not get version bumps when we do a develop merge of hotfixes. I'd propose we merge master into develop instead of merging the hotfix branch into develop instead to fix this, or just don't worry about the develop version.

## Type of change


## How should this be tested?

- Can see it in my personal repo: https://github.com/timjrobinson/frontend-v2/tree/master after I merged https://github.com/timjrobinson/frontend-v2/pull/14, deploy branch is the same as master and is pointing to that latest tag. 
- Can fork this repo to test in your own personal version and see how it works. 

## Checklist:
